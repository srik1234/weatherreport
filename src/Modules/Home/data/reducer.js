

const initialState = {
    cityHistoricalData: [],
    selectedCity: null,
    loading: false,
    error: false,
};
export const cityWeatherReducer = (state = initialState, action) => {
    switch (action.type) {
        case 'FETCHING_CITIES_WEATHER':
            return { ...state, data: action.position, loading: false }
        case 'UPDATE_CITY_HISTORICALDATA':
            return { ...state, cityHistoricalData: action.data, loading: false }
        case 'SET_SELECTED_CITY':
            return { ...state, selectedCity: action.selectedCity, loading:true }
        default:
            return state;
    }
}
