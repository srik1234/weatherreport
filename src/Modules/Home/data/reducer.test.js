import * as reducer from './reducer';


const initialState = {
    cityHistoricalData: [],
    selectedCity: null,
    loading: false,
    error: false,
};

describe('cityweather reducer', () => {
    it('should return the initial state', () => {
        expect(reducer.cityWeatherReducer(undefined, {})).toMatchSnapshot()
    })
    it('should handle cityWeatherRequest', () => {
        expect(
            reducer.cityWeatherReducer(initialState,
                {
                    type: 'UPDATE_CITY_HISTORICALDATA'
                })
        ).toMatchSnapshot()
    })
});