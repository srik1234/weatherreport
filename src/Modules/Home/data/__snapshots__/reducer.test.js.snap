// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cityweather reducer should handle cityWeatherRequest 1`] = `
Object {
  "cityHistoricalData": undefined,
  "error": false,
  "loading": false,
  "selectedCity": null,
}
`;

exports[`cityweather reducer should return the initial state 1`] = `
Object {
  "cityHistoricalData": Array [],
  "error": false,
  "loading": false,
  "selectedCity": null,
}
`;
